### ISSUE TITLE
[docs] Developer docs

### ISSUE THREAD

**Issue by behdad** [harfbuzz/harfbuzz#3405]
As far as I know, `docs/repacker.md` and the new `docs/serializer.md` are unlinked, not even part of the dist tarball. Should be fixed.

**Comment by khaledhosny** [comment]
There is no simple way to integrate markdown into the DocBook based gtk-doc workflow. The easiest is probably to use something like pandoc to generate DockBook from the markdown and integrate that into the documentation (possibly while keeping the markdown files as the editable source).

**Comment by matthiasclasen** [comment]
The more radical approach is to switch the docs to use gi-docgen. Thats what the gtk stack and much of gnome has switched to. Will take some rewriting though

**Comment by khaledhosny** [comment]
I looked at gi-docgen generated output, but I find it rather confusing to browse for a C API and seems very introspection-centeric.

**Comment by matthiasclasen** [comment]
Yeah, one of the main ideas was to force docs to be all introspection based. It can be limiting for the docs, but at the same time it makes the introspection api much more visible and therefore, hopefully, better maintained. May not be the right thing for harfbuzz, of course.

**Comment by n8willis** [comment]
I would be thrilled to ditch the XML and never look back. Markdown has its share of frustrating quirks, but the real win is that it retains readability in original format. I don't think that DocBook ever even considered that. Not having "learn how to use the document generator and its stand-alone format" as prerequisites to adding/updating text would be a long-term improvement.

That being said, I don't have a lot to weigh in on about the doc generator options if they need to be built into the overall build system. Last couple times I tried Meson I couldn't get it to cooperate.

**Comment by twardoch** [comment]
One way could be to use .ipynb and something like https://jupyterbook.org/

Basically .ipynb is a JSON which includes cells (blocks, kind of like WordPress Gutemberg blocks), and each cell can be interpreted with a different interpreter. So one could mix autogenerated content with manual content written in different markups/languages. 

In addition, at least for shell & Python usages, one can do interactive things when one opens the .ipynb in an environment like Google Colab. See 

https://github.com/twardoch/twardoch.github.io/blob/master/Test_Google_Colab_Jupyter_Notebook.ipynb

for an example (click the Open in Colab link)

**Comment by behdad** [comment]
> As far as I know, `docs/repacker.md` and the new `docs/serializer.md` are unlinked, not even part of the dist tarball. Should be fixed.

We should at least link them from README.md. That's where our whatever developer docs we have are linked from.
